src/lib/ui/SupportedDevices/SupportedDevices.tsx

```diff
@@ -43,22 +105,43 @@ export function SupportedDevices() {
           </div>
         )}
 
-        {!isLoading &&
-          !isError &&
-          data?.data?.device_models !== null &&
-          Array.isArray(data?.data?.device_models) && (
-            <table className='seam-supported-devices-table'>
-              <SupportedDevicesHeader />
-              <tbody>
-                {data.data.device_models.map((deviceModel) => (
+        {!isLoading && !isError && data?.data?.device_models !== null && (
+          <table className='seam-supported-devices-table'>
+            <SupportedDevicesHeader />
+            <tbody>
+              {deviceModels.length !== 0 &&
+                deviceModels.map((deviceModel, index) => (
                   <SupportedDeviceRow
-                    key={deviceModel.manufacturer_model_id}
+                    key={`${index}:${deviceModel.manufacturer_model_id}`}
                     deviceModel={deviceModel}
                   />
                 ))}
-              </tbody>
-            </table>
-          )}
+
+              {deviceModels.length === 0 && (
+                <tr className='seam-supported-devices-table-message-row'>
+                  <td colSpan={6}>
+                    <div className='seam-supported-devices-table-message'>
+                      {filterValue.length === 0 ? (
+                        <p>No device models found.</p>
+                      ) : (
+                        <>
+                          <p>No device models matched your search.</p>
+                          <Link
```

Comment:
Can we replace this with `a` or otherwise?

Reply:
Replaced with small `Button` from this project.